templates/student_rent.html

```
@@ -200,57 +172,46 @@ <h2 class="h6 mb-0 d-flex align-items-center justify-content-between">
         </div>
         {% else %}
         <div class="alert alert-info mb-3">
-            <strong>Grace Period:</strong> Pay by {{ grace_end_date_for_status.strftime('%B %d, %Y') }} to avoid late
-            fee.
+            <strong>Grace Period:</strong> Pay by {% if status.grace_end %}{{ status.grace_end.strftime('%B %d, %Y') }}{% else %}the grace deadline{% endif %} to avoid late fee.
         </div>
         {% endif %}
 
         {% if not status.is_paid %}
-            {% if settings.allow_incremental_payment or checking_balance >= status.remaining_amount %}
+            {% if checking_balance >= status.remaining_amount %}
```

HTMLHint (1.9.2)

[error] Special characters must be escaped : [ > ].

(spec-char-escape)